otter_welcome_buddy/cogs/englishclub_reminder.py

```diff
@@ -0,0 +1,132 @@
+import os
+import re
+
+import discord
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from discord import Embed
+from discord import TextChannel
+from discord.ext import commands
+from discord.ext.commands import Bot
+from discord.ext.commands import Context
+
+from otter_welcome_buddy.common.constants import CronExpressions
+from otter_welcome_buddy.common.utils.dates import DateUtils
+from otter_welcome_buddy.common.utils.types.common import DiscordChannelType
+from otter_welcome_buddy.formatters import englishclub_message
+
+
+def is_allowed(ctx: Context) -> bool:
+    """
+    Return if a user is allowed to run a command or not
+    """
+    return ctx.author.id == int(os.environ["USER_ONE_ID"]) or ctx.author.id == int(
+        os.environ["USER_TWO_ID"],
+    )
+
+
+class English(commands.Cog):
+    """
+    English command events, where notifications about English sessions
+    Commands:
+        englishclub start:     Start cronjob for timeline messages
+        englishclub stop:      Stop cronjob for timeline messages
+    """
+
+    def __init__(self, bot: Bot, messages_dependency: type[englishclub_message.Formatter]):
+        self.bot: Bot = bot
+        self.messages_formatter: type[englishclub_message.Formatter] = messages_dependency
+        self.scheduler: AsyncIOScheduler = AsyncIOScheduler()
+        self.__configure_scheduler()
+
+    @commands.group(
+        brief="Commands related to English Club!",
+        invoke_without_command=True,
+        pass_context=True,
+    )
+    async def englishclub(self, ctx: Context) -> None:
+        """
+        English Club will setup an English session
+        """
+        await ctx.send_help(ctx.command)
+
+    @englishclub.command(brief="Start the cronjob for the messages")  # type: ignore
+    async def start(self, _: Context) -> None:
+        """Command to interact with the bot and start cron"""
+        self.__configure_scheduler()
+
+    @englishclub.command(brief="Stop the cronjob for the messages")  # type: ignore
+    async def stop(self, _: Context) -> None:
+        """Command to interact with the bot and stop cron"""
+        self.scheduler.remove_job("job_id")
+
+    @commands.command(brief="Schedule an English session")
+    @commands.check(is_allowed)
+    async def schedule_session(self, ctx: Context, hour: str) -> None:
+        """Schedule the English Club session"""
+
+        pattern = r"^\d{1,2}:\d{2}(AM|PM)$"
```

Comment:
pattern it's a pretty vague name, what its the pattern about? Also it's a good practice to explain step by step what the regex does with comments, it's difficult to read regex on the wild.